x-pack/plugins/fleet/server/routes/agent_policy/index.ts

```diff
@@ -28,85 +28,100 @@ import {
   downloadFullAgentPolicy,
 } from './handlers';
 
-export const registerRoutes = (routers: { superuser: FleetRouter; fleetSetup: FleetRouter }) => {
+export const registerRoutes = (router: FleetAuthzRouter) => {
   // List - Fleet Server needs access to run setup
-  routers.fleetSetup.get(
+  router.get(
     {
       path: AGENT_POLICY_API_ROUTES.LIST_PATTERN,
       validate: GetAgentPoliciesRequestSchema,
-      // Disable this tag and the automatic RBAC support until elastic/fleet-server access is removed in 8.0
-      // Required to allow elastic/fleet-server to access this API.
-      // options: { tags: [`access:${PLUGIN_ID}-read`] },
+      fleetAllowFleetSetupPrivilege: true,
+      fleetAuthz: {
+        fleet: { all: true },
+      },
```

Comment:
Instead of making `fleetAllowFleetSetupPrivilege` a separate flag, I think we should add a dedicated field to `FleetAuthz.fleet` like `FleetAuthz.fleet.readAgentPolicies` and then use
```suggestion
      fleetAuthz: {
        fleet: { readAgentPolicies: true },
      },
```

This would allow us to move all of the authz decision logic into `calculateAuthz` to keep things very simple for unit testing.